### ISSUE TITLE
ContentDialog is not read in the right order by Narrator

### ISSUE THREAD

**Issue by lyf6lyf** [microsoft/microsoft-ui-xaml#5909]
**Describe the bug**
`ContentDialog` is not read in the right order by Narrator. It is more obvious if our dialog is complecated.

**Steps to reproduce the bug**
We can repro in XAML Control Gallery
1. Open XAML Control Gallery
2. Open `Dialogs` and Flyouts` -> `ContentDialog`
3. Open Narrator
4. Press tab several time to set the focus on the Show Dialog Button
5. Press Enter to open the dialog
6. The elements are read in below order
![image](https://user-images.githubusercontent.com/6219635/133872656-b6fde916-182f-4a95-94ad-753a1862d926.png)

**Expected behavior**
The dialog should be read from top to the buttom.

**Version Info**

NuGet package version: 
Microsoft.UI.Xaml 2.7.0

| Windows version                    | Saw the problem? |
| :--------------------------------- | :-------------------- |
| Insider Build (22460.1000)              |  Yes   |
| May 2021 Update (19043)            |    |
| October 2020 Update (19042)        |    |
| May 2020 Update (19041)            |    |
| November 2019 Update (18363)       |    |
| May 2019 Update (18362)            |    |
| October 2018 Update (17763)        |    |
| April 2018 Update (17134)          |    |
| Fall Creators Update (16299)       |    |
| Creators Update (15063)            |    |

| Device form factor | Saw the problem? |
| :----------------- | :--------------- |
| Desktop            | Yes |
| Xbox               |  |
| Surface Hub        |  |
| IoT                |  |

**Additional context**

**Comment by ranjeshj** [comment]
@lyf6lyf can you provide the example where this is problematic ? Narrator is reading the contents of the entire dialog because we have AccessibilityProperties.IsDialog set to true for the dialog. The contents in the UIA  tree seem to be in the correct order, however focus is on the checkbox when we load, so perhaps it is reading the header, then the actionable content, then the rest which seems reasonable.

**Comment by lyf6lyf** [comment]
@ranjeshj our real example is something like this
![image](https://user-images.githubusercontent.com/6219635/135377491-84456002-5670-436a-8023-06a70a4c87e7.png)
Yes the reading order is interrupted by the focused control, but it makes user confused.

Would it be better if narrator read the default focused control after all other text has been read?

**Comment by zadjii-msft** [comment]
@carlos-zamora wait haven't we seen this in the Terminal before?

**Comment by carlos-zamora** [comment]
Yeah, this repros in the About dialog in Windows Terminal:
<img width="253" alt="image" src="https://github.com/microsoft/microsoft-ui-xaml/assets/11050425/c6b05e7c-c179-4ea7-9787-b5035ee9710d">

The order is...
1. About
2. Source code (focused element)
3. text above the hyperlink
4. all of the other hyperlinks

**Comment by zadjii-msft** [comment]
Ah yea, this was https://github.com/microsoft/terminal/issues/11295

@bpulliam can we get this re-opened? Then we can dupe our bug to this one.
